storage/retention.go

```diff
@@ -111,6 +120,16 @@ func (s *retentionEnforcer) expireData(ctx context.Context, buckets []*influxdb.
 	logger, logEnd := logger.NewOperation(ctx, s.logger, "Data deletion", "data_deletion")
 	defer logEnd()
 
+	// Snapshot to clear the cache to reduce write contention.
+	span, _ := tracing.StartSpanFromContextWithOperationName(ctx, "retention_snapshot")
```

Comment:
The correct way to trace `WriteSnapshot()` is to simply pass the context object `ctx` in to the `WriteSnapshot()` method. As I see things, this is not the right place to create a new context (`context.Background()`).

Looks like `Engine.WriteSnapshot()` [does create a span](https://github.com/influxdata/idpe/blob/8d40e7b3f788e2ef79d7fa7e2a996b10500a8467/vendor/github.com/influxdata/influxdb/tsdb/tsm1/engine.go#L780), so this should be simple.

Reply:
FWIW, `s.Engine.DeleteBucketRange` is wrapped in a span because it doesn't take a context object (which needs to be fixed; all IO operations should be traced).